src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
While an oversimplification, all a Try is basically an Either[Throwable, Seq[StorageObject]]. Our sequence method is effectively treating it as an Either[String, Seq[StorageObject]] which looks a heck of a lot like the ValidationNEL[String, Seq[StorageObject]] which you're decrying.

The point is that if you want a data structure which allows you to accumulate error messages vs fail-fast, that already exists (i.e. an applicative). We've reinvented a wheel. 

As an analogy would you suggest we always use double instead of ints? Sure we could wrap methods around double to make sure we get int behavior in our specific cases, and the code would probably be pretty simple as well.